proxy/src/control/destination/mod.rs

```diff
@@ -93,21 +95,7 @@ pub struct Metadata {
     dst_labels: Option<DstLabels>,
 
     /// How to verify TLS for the endpoint.
-    tls_identity: Option<Arc<TlsIdentity>>,
-}
-
-/// How to verify TLS for an endpoint.
-///
-/// XXX: This currently derives `PartialEq  and `Eq`, which is not entirely
-/// correct, as domain name equality ought to be case insensitive. However,
-/// `Metadata` must be `Eq`.
-#[derive(Debug, PartialEq, Eq)]
-pub enum TlsIdentity {
```

Comment:
Thanks for moving this to the `tls` module, it probably should have been there in the first place (but that didn't exist at the time).